resources/ec2-verified-access-endpoint.go

```diff
@@ -0,0 +1,104 @@
+package resources
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+
+	"github.com/ekristen/libnuke/pkg/registry"
+	"github.com/ekristen/libnuke/pkg/resource"
+	"github.com/ekristen/libnuke/pkg/types"
+
+	"github.com/ekristen/aws-nuke/v3/pkg/nuke"
+)
+
+const EC2VerifiedAccessEndpointResource = "EC2VerifiedAccessEndpoint"
+
+func init() {
+	registry.Register(&registry.Registration{
+		Name:     EC2VerifiedAccessEndpointResource,
+		Scope:    nuke.Account,
+		Resource: &EC2VerifiedAccessEndpoint{},
+		Lister:   &EC2VerifiedAccessEndpointLister{},
+	})
+}
+
+type EC2VerifiedAccessEndpointLister struct{}
+
+func (l *EC2VerifiedAccessEndpointLister) List(ctx context.Context, o interface{}) ([]resource.Resource, error) {
+	opts := o.(*nuke.ListerOpts)
+
+	svc := ec2.NewFromConfig(*opts.Config)
+
+	params := &ec2.DescribeVerifiedAccessEndpointsInput{}
+	resources := make([]resource.Resource, 0)
+
+	for {
+		resp, err := svc.DescribeVerifiedAccessEndpoints(ctx, params)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, endpoint := range resp.VerifiedAccessEndpoints {
+			resources = append(resources, &EC2VerifiedAccessEndpoint{
+				svc:      svc,
+				endpoint: &endpoint,
```

Comment:
We've moved away from embedding full structs because of memory and because the tooling now has automatic property documentation generation, etc.

For example https://github.com/ekristen/aws-nuke/blob/main/resources/bedrock-flow-alias.go#L71-L76 followed by https://github.com/ekristen/aws-nuke/blob/main/resources/bedrock-flow-alias.go#L85-L87

This allows for setting the properties once on the struct and then the library takes care of it. Tags are automatically supported to https://github.com/ekristen/aws-nuke/blob/959fd5b479e217a878427b1cf9162148b36144ec/resources/iam-user.go#L40-L51